### ISSUE TITLE
Screen Reader: Reports: The icon button near search field is announced without a name

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77464]
## Action Performed:
Prerequisite:
1. User must be logged in

1. Using Android TalkBack or iOS VoiceOver, open the application.
2. Once the 'Inbox' screen opens, swipe right to focus on the 'Reports' button at the bottom of the screen and double-tap to activate it.
3. Once the 'Reports' screen opens, swipe right to focus on the 'Menu' button, which is available next to the search field.
4. Note how the control is announced.

## Expected Result:
The icon button should have a meaningful accessible name, e.g.:
- 'reports menu'

## Actual Result:
The icon button next to the search field is announced by the screen reader without any accessible or programmatic name. As a result, users cannot understand the purpose or function of the button.

## Area issue was found in:
Reports

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure to use the accessibilityLabel to announce the name of the button, e.g.:
- 'reports menu'
https://reactnative.dev/docs/accessibility#accessibilitylabel

## Environment:
Xiaomi Redmi 11 Prime 5G - Android 13, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos

<img width="304" height="632" alt="Image" src="https://github.com/user-attachments/assets/8e2bdc3f-12fa-4fec-894e-f83108960357" />

https://github.com/user-attachments/assets/207dfdef-1f62-406b-a163-a712ef2a9a80

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7007047
**Issue reported by:** NA
**Slack conversation:** NA
